public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -114,60 +219,369 @@ function FolderGroupAndEvaluationInterval({
+                  <Switch
+                    id="pause-alert"
+                    onChange={(value) => {
+                      setValue('isPaused', value.currentTarget.checked);
+                    }}
+                    value={Boolean(isPaused)}
+                  />
+                  <label htmlFor="pause-alert" className={styles.switchLabel}>
+                    <Trans i18nKey="alerting.rule-form.pause.label">Pause evaluation</Trans>
+                    <Tooltip placement="top" content={pauseContentText} theme={'info'}>
+                      <Icon tabIndex={0} name="info-circle" size="sm" className={styles.infoIcon} />
+                    </Tooltip>
+                  </label>
+                </Stack>
+              )}
+              name="isPaused"
+            />
+          </Field>
+        )}
+      </Stack>
+      {isGrafanaAlertingRule && (
+        <>
+          <CollapseToggle
+            isCollapsed={!showErrorHandling}
+            onToggle={(collapsed) => setShowErrorHandling(!collapsed)}
+            text="Configure no data and error handling"
+          />
+          {showErrorHandling && (
+            <>
+              <NeedHelpInfoForConfigureNoDataError />
+              <Field htmlFor="no-data-state-input" label="Alert state if no data or all values are null">
+                <Controller
+                  render={({ field: { onChange, ref, ...field } }) => (
+                    <GrafanaAlertStatePicker
+                      {...field}
+                      inputId="no-data-state-input"
+                      width={42}
+                      includeNoData={true}
+                      includeError={false}
+                      onChange={(value) => onChange(value?.value)}
+                    />
+                  )}
+                  name="noDataState"
+                />
+              </Field>
+              <Field htmlFor="exec-err-state-input" label="Alert state if execution error or timeout">
+                <Controller
+                  render={({ field: { onChange, ref, ...field } }) => (
+                    <GrafanaAlertStatePicker
+                      {...field}
+                      inputId="exec-err-state-input"
+                      width={42}
+                      includeNoData={false}
+                      includeError={true}
+                      onChange={(value) => onChange(value?.value)}
+                    />
+                  )}
+                  name="execErrState"
+                />
+              </Field>
+            </>
+          )}
+        </>
       )}
-    </div>
+    </RuleEditorSection>
   );
 }
 
-function ForInput({ evaluateEvery }: { evaluateEvery: string }) {
+function EvaluationGroupCreationModal({
+  onClose,
+  onCreate,
+  groupfoldersForGrafana,
+}: {
+  onClose: () => void;
+  onCreate: (group: string, evaluationInterval: string) => void;
+  groupfoldersForGrafana?: RulerRulesConfigDTO | null;
```

Comment:
Same question as above. Why do we want to allow null here?

Reply:
same